packages/backend/src/factcollectors/branchcount.ts

```diff
@@ -57,86 +82,122 @@ export class BranchCountFactCollector implements FactCollector {
-    _params?: { factRefs?: FactRef[]; refresh?: FactRef[] },
-  ): Promise<Fact[]> {
+    _params?: { factRefs?: FactRef[]; refresh?: FactRef[] }
+  ) {
     try {
       const factRef: FactRef = stringifyFactRef({
         name: 'branch_count',
         scope: 'default',
-        source: 'branch',
+        source: this.id,
       });
-      return Promise.all(
+      const results = await Promise.all(
         entities
-          .filter(entity => isScmEntity(entity))
-          .map(async entity => {
-            const entityRef = stringifyEntityRef(entity);
-            const entityScmUrl = getEntityScmUrl(entity);
-            const gitUrl = parseGitUrl(entityScmUrl);
-
-            const { token } = await this.#credentialsProvider.getCredentials({ url: entityScmUrl });
-            const response = await graphql(
-              `
-              query numBranches($owner: String!, $repo: String!) {
-                repository(owner: $owner, name: $repo) {
-                  refs(first: 0, refPrefix: "refs/heads/") {
-                    totalCount
-                  }
-                }
-              }
-            `,
-              {
-                owner: gitUrl.owner,
-                repo: gitUrl.name,
-                headers: {
-                  authorization: 'Bearer ' + token,
-                },
-              }
-            ) as GraphQlQueryResponseData;
-            
-            console.log("BranchCountFactCollector: " + gitUrl.owner + ": " + gitUrl.name + ": " + "Total Count: "
-            + response["repository"]["refs"]["totalCount"]);
-
-            return this.buildFact(entityRef, factRef, response["repository"]["refs"]);
-          }),
+          .filter((entity) => isScmEntity(entity))
+          .map((entity) => this.collectData(entity, factRef))
       );
+
+      return results.filter((result): result is Fact => result !== null);
     } catch (e) {
       this.#logger.error(`Failed to collect branch data with error: ${e}`);
       return Promise.reject([]);
     }
   }
 
-  getCollectionConfigs(): Promise<CollectionConfig[]> {
+  async collectData(entity: Entity, factRef: string) {
+    const entityRef = stringifyEntityRef(entity);
+    const entityScmUrl = getEntityScmUrl(entity);
+    const gitUrl = parseGitUrl(entityScmUrl);
+
+    const { token } = await this.#credentialsProvider.getCredentials({
+      url: entityScmUrl,
+    });
+    const {
+      repository: { refs: totalCount },
+    } = (
+      await graphql<GraphQlQueryResponseData>(
+        `
+          query numBranches($owner: String!, $repo: String!) {
+            repository(owner: $owner, name: $repo) {
+              refs(first: 0, refPrefix: 'refs/heads/') {
+                totalCount
+              }
+            }
+          }
+        `,
+        {
+          owner: gitUrl.owner,
+          repo: gitUrl.name,
+          headers: {
+            authorization: `Bearer ${token}`,
+          },
+        }
+      )
+    ).catch((e) => {
```

Comment:
This results in the following error:
> TypeError: (intermediate value).catch is not a function

Can you wrap the GraphQL call in a try/catch block?